source-hubspot-native/source_hubspot_native/api.py

```diff
@@ -201,6 +355,41 @@ async def fetch_changes(
 
     if recent:
         yield recent[-1][0]
+    else:
+        return
+
+
+async def fetch_changes_no_batch(
```

Comment:
Can you comment more on what this function is for? Coming in without a deep understanding of the pre-existing code, I am struggling to understand why this is needed as an alternative to `fetch_changes`. I see that it does things a little differently and that it doesn't have that final `fetch_batch_with_associations` part, but I need a little help understanding what the reasoning for that is.